### PR TITLE
Add namespace to cmake options

### PR DESCRIPTION
**Pull request by hjmjohnson**
Provide a more clean interface for selecting options to build.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@thewtex @dzenanz This is the last change I'd like to make to VXL before changing ITK to require at least VXL version 2.0.0 for building.  It hides some of the VXL components from the main screen (mark_as_advanced) and it clearly identifies VXL options as separate from ITK options.

**Comment by hjmjohnson** [comment]
@decrispell Thanks.  My OCD for fixing the VXL code is subsiding :)  I hope that only small changes changes are needed from here on out.   There are some annoying warnings on the dashboard. And there are 2 tests that fail when building absolutely everything (but those two were failing before I started.)

File: config/cmake/Modules/VXLStandardOptions.cmake [hunk 1]
```diff
@@ -32,7 +32,7 @@ if( WIN32 )
 else()
   option( BUILD_SHARED_LIBS "Build shared libraries." OFF)
 endif()
-set( VXL_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS} )
+set( BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS} )
```
Comment (dzenanz) [inline config/cmake/Modules/VXLStandardOptions.cmake]: This line is no longer necessary.

**Comment by dzenanz** [comment]
Thanks for maintenance!

**Pull request merged**